tests/test_lesion.py

```diff
@@ -0,0 +1,211 @@
+import pytest
+from liver_imaging_analysis.models.lesion_segmentation import (
+    LesionSegmentation,
+    segment_lesion,
+    train_lesion,
+)
+from liver_imaging_analysis.engine.transforms import MorphologicalClosing
+from liver_imaging_analysis.models.liver_segmentation import segment_liver
+from liver_imaging_analysis.engine.config import config
+import torch
+import nibabel as nib
+from liver_imaging_analysis.engine.dataloader import DataLoader, Keys
+import numpy as np
+from liver_imaging_analysis.engine.utils import VolumeSlicing
+from monai.transforms import (
+    Compose,
+    EnsureChannelFirstD,
+    LoadImageD,
+    RandFlipd,
+    RandRotated,
+    ToTensorD,
+    RandAdjustContrastd,
+    RandZoomd,
+    ActivationsD,
+    AsDiscreteD,
+    FillHolesD,
+    ScaleIntensityRanged,
+    RemoveSmallObjectsD,
+)
+
+
+@pytest.fixture
+def lesion_obj():
+    lesion_obj = LesionSegmentation()
+    return lesion_obj
+
+
+@pytest.mark.parametrize("transform_name", [("2d_ct_transform")])
+def test_get_pretraining_transforms(transform_name, lesion_obj):
+    """
+    Tests get_pretraining_transforms function.
+    verifies the functionality by comparing the output transforms with the expected transforms based on the provided transform_name.
+
+    Parameters:
+    ----------
+    transform_name (string): "2d_ct_transform".
+    lesion_obj (object): The lesion object that provides the get_pretraining_transforms method.
+    ----------
+    """
+
+    output_transform = lesion_obj.get_pretraining_transforms(transform_name)
+    assert isinstance(output_transform, Compose)
+
+    if transform_name == "2d_ct_transform":
+        expected_transfom = [
```

Comment:
what if you change the transforms in the lesions module? should the test fail then? maybe this test should just check that a compose object containing dictionary transforms is returned (by checking that each of them is an instance of MapTransform, for example)

Reply:
Is it necessary to also check the types of transforms?